libkrun/libkruntime.c

```diff
@@ -37,86 +79,254 @@ clock_gettime_monotonic()
 }
 
 /*
- * JNI Implementation -- Optionally compiled in
+ * JNI wrappers -- Optionally compiled in
  */
 #ifdef WITH_JAVA
+JNIEXPORT void JNICALL
+Java_IterationsRunner_JNI_1libkruntime_1init(JNIEnv *e, jclass c) {
+    libkruntime_init();
+}
+
+JNIEXPORT void JNICALL
+Java_IterationsRunner_JNI_1libkruntime_1done(JNIEnv *e, jclass c) {
+    libkruntime_done();
+}
 
 JNIEXPORT jdouble JNICALL
 Java_IterationsRunner_JNI_1clock_1gettime_1monotonic(JNIEnv *e, jclass c) {
     return (jdouble) clock_gettime_monotonic();
 }
 
 JNIEXPORT jlong JNICALL
-Java_IterationsRunner_JNI_1read_1ts_1reg_1start(JNIEnv *e, jclass c)
+Java_IterationsRunner_JNI_1read_1core_1cycles(JNIEnv *e, jclass c)
 {
-    return read_ts_reg_start();
+    return read_core_cycles();
 }
+#endif
 
-JNIEXPORT jlong JNICALL
-Java_IterationsRunner_JNI_1read_1ts_1reg_1stop(JNIEnv *e, jclass c)
+#if defined(__linux__) && !defined(TRAVIS)
+int
+open_msr_node(int core, int flags)
 {
-    return read_ts_reg_stop();
+    char path[MAX_MSR_PATH];
+    int msr_node;
+
+    /* Note this is not the default msr(4) device! */
```

Comment:
Maybe be clearer in the comment that we're tracking `rmsr`, not `msr` (at the moment, the comment requires the reader to know that).